src/objects/set.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2018 David Grunwald, Georges Racinet
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this
+// software and associated documentation files (the "Software"), to deal in the Software
+// without restriction, including without limitation the rights to use, copy, modify, merge,
+// publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons
+// to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+// PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+// FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use ffi;
+use python::{Python, PythonObject};
+use conversion::ToPyObject;
+use objects::PyObject;
+use err::{self, PyResult, PyErr};
+use std::{mem, collections, hash, cmp, ptr};
+
+/// Represents a Python `set`.
+pub struct PySet(PyObject);
+
+pyobject_newtype!(PySet, PySet_Check, PySet_Type);
+
+impl PySet {
+    /// Creates a new set from any iterable
+    ///
+    /// Corresponds to `set(iterable)` in Python.
+    pub fn new<I>(py: Python, iterable: I) -> PyResult<PySet> where I: ToPyObject {
+        iterable.with_borrowed_ptr(py, |iterable| unsafe {
+            err::result_cast_from_owned_ptr(py, ffi::PySet_New(iterable))
+        })
+    }
+
+    /// Creates an empty set
+    ///
+    /// Corresponds to `set()` in Python
+    #[inline]
+        pub fn empty(py: Python) -> PyResult<PySet> {
+        unsafe {
+            err::result_cast_from_owned_ptr(py,
+                ffi::PySet_New(ptr::null_mut()))
+        }
+    }
+
+    /// Empty an existing set of all values.
+    #[inline]
+    pub fn clear(&self, py: Python) -> PyResult<()> {  // NOCOMMIT check CPython doc for signature explanation (dict's is different)
```

Comment:
Have you got a resolution for this?

Reply:
Yes, I checked the signature difference, it's perfectly correct.
(this is usually catched by my precommt hooks if I don't forget to activate them)